db/data_migrations/20241024131255_migrate_data_request_areas.rb

```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class MigrateDataRequestAreas < ActiveRecord::DataMigration
+  def data_request_area_type_for(request_type)
+    case request_type
+    when *DataRequest::BRANSTON_DATA_REQUEST_TYPES
+      "branston"
+    when *DataRequest::BRANSTON_REGISTRY_DATA_REQUEST_TYPES
+      "branston_registry"
+    when *DataRequest::MAPPA_DATA_REQUEST_TYPES
+      "mappa"
+    when *DataRequest::PRISON_DATA_REQUEST_TYPES + %w[cctv_and_bwcf] - %w[other]
+      "prison"
+    when *DataRequest::PROBATION_DATA_REQUEST_TYPES + %w[court] - %w[other]
+      "probation"
+    when "other"
+      "other_department"
+    end
+  end
+
+  def up
+    # loop through the DataRequest records
+    DataRequest.find_each do |request|
+      # Get the data_request_area_type for this request
+      data_request_area_type = data_request_area_type_for(request.request_type)
+
+      Rails.logger.debug "Finding DataRequestArea with user_id: #{request.user_id}, case_id: #{request.case_id}, contact_id: #{request.contact_id.presence}"
+
+      data_request_area = DataRequestArea.new(
+        user_id: request.user_id,
+        case_id: request.case_id,
+        data_request_area_type:,
+        contact_id: request.contact_id,
+        location: request.location.presence,
+        )
+
+      # Prevents existing data request commissioning documents being overwritten during the migration
+      if request.commissioning_document.present?
```

Comment:
commissioning_document has been removed as an association from data_request in the main PR, so you will have to restore the association there, or perform a query on CommissioningDocument.

Reply:
Yes I added it to my local setup, I wasnt sure whether it needed to stay. I guess it does if the new flow is merged before the migration is done